docusaurus/docs/e2e-test-a-plugin/locating-elements.md

```diff
@@ -0,0 +1,43 @@
+---
+id: locating-elements
+title: Locating elements for E2E tests
+description: How to locate elements on the page
+draft: true
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - e2e
+  - example test
+sidebar_position: 30
+---
+
+# Locating elements for E2E tests
+
+Ever since Grafana 7.0.0, end-to-end tests in Grafana have relied on selectors defined in the [`@grafana/e2e-selectors`](https://github.com/grafana/grafana/tree/main/packages/grafana-e2e-selectors) package to locate elements in the Grafana UI. 
+
+Locating Grafana UI elements can be challenging because the selector may be defined on either the `aria-label` attribute or the `data-testid` attribute. In the beginning, the selectors used the `aria-label` attribute, but now most selectors have been migrated to use the `data-testid` attribute instead.
+
+# Locating Grafana UI elements
+
+All pages defined by `@grafana/plugin-e2e` expose a `getByTestIdOrAriaLabel` method that returns a Playwright locator that resolves to element(s) selected using the right HTML attribute. Whenever you want to resolve a Playwright locator based on a Grafana UI selector, you should always use this method.
```

Comment:
> All pages defined by `@grafana/plugin-e2e`

nit: this makes me ask "which pages do we define" and would want to see a list.

Reply:
```suggestion
All pages defined by `@grafana/plugin-e2e` expose a `getByTestIdOrAriaLabel` method that returns a Playwright locator that resolves to element(s) selected, using the appropriate HTML attribute as defined on the element. Whenever you want to resolve a Playwright locator based on a Grafana UI selector, you should always use this method.
```

maybe clearer, maybe not. Just a suggestion as I felt "right" was ambiguous.

Reply:
>nit: this makes me ask "which pages do we define" and would want to see a list.

yeah I see what you mean. it's all the files that have the `Page` suffix in [this](https://github.com/grafana/plugin-tools/tree/main/packages/plugin-e2e/src/models) folder. Would be great if we had an `API reference > Models` section in the docs that was auto generated from the code. Then we could refer to that from within this article. Maybe we can get back to this once we have ways of auto-generating docs from tsdoc? (gonna bring this up with the rest of the team, but doesn't feel like it's prioritized right now)

Reply:
wdyt about just adding the link to that folder on "All `[pages](https://github.com/grafana/plugin-tools/tree/main/packages/plugin-e2e/src/models)` defined..."? I agree with the future direction, but this feels a reasonable stop gap

Reply:
👍